astropy/convolution/boundary_none.pyx

```diff
@@ -59,17 +59,29 @@ def convolve1d_boundary_none(np.ndarray[DTYPE_t, ndim=1] f,
     # GIL acquired again here
     return conv
 
-
-@cython.boundscheck(False)  # turn off bounds-checking for entire function
+@cython.boundscheck(False)
 def convolve2d_boundary_none(np.ndarray[DTYPE_t, ndim=2] f,
                              np.ndarray[DTYPE_t, ndim=2] g,
-                             bint normalize_by_kernel):
+                             bint nan_interpolate):
 
     if g.shape[0] % 2 != 1 or g.shape[1] % 2 != 1:
         raise ValueError("Convolution kernel must have odd dimensions")
 
     assert f.dtype == DTYPE and g.dtype == DTYPE
 
+    if nan_interpolate:
```

Comment:
By placing this condition here and then calling the function below with True/False literals, along with the called function being inlined, the code will be expanded and the conditions removed.